lib/initInstance.js

```diff
@@ -104,6 +104,27 @@ if (createJson) {
 //740 specifically, to keep server config secure
 }
 
+function generateComponentJson() {
+  let componentJsonContent;
+  try {
+    componentJsonContent = require(versionLocation);
+  } catch (e) {
+    componentJsonContent = {};
+    //doesnt exist, create new
+  }
+  let currentManifestJson;
+  try {
+    currentManifestJson = fs.readFileSync(path.join(__dirname, "../../../../../manifest.json"));
+  } catch (e) {
+    if (e.code == 'ENOENT') {
+      createJson = true;
+    } else {
+      console.log('Warning: Could not read manifest.json, error='+e.message);
+    }
+  }
+  componentJsonContent.version = currentManifestJson.version;
```

Comment:
initInstance runs in a dev environment too, where manifest.json does not exist.
this line is outside the try-catch, so it will throw an exception and fail.
You should initialize currentManifestJson to {version:"0.0.0.0"} so that dev environments dont have this issue.